azurerm/internal/services/analysisservices/analysis_services_server_resource.go

```diff
@@ -252,6 +252,32 @@ func resourceArmAnalysisServicesServerUpdate(d *schema.ResourceData, meta interf
 		return err
 	}
 
+	serverResp, err := client.GetDetails(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(serverResp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error retrieving Analysis Services Server %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if serverResp.State != analysisservices.StateSucceeded && serverResp.State != analysisservices.StatePaused {
+		return fmt.Errorf("Error updating Analysis Services Server %q (Resource Group %q): State must be either Succeeded or Paused", id.Name, id.ResourceGroup)
```

Comment:
it'd be worth adding the current state here

```suggestion
		return fmt.Errorf("Error updating Analysis Services Server %q (Resource Group %q): State must be either Succeeded or Paused but got %q", id.Name, id.ResourceGroup, serverResp.State)
```